### PR TITLE
fix 3546: If you have invalid code in you freemarker the dataexplorer…

### PR DESCRIPTION
**Pull request by bartcharbon**
… shows the default template, and logs no error

### CODE REVIEW

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java [hunk 1]
```diff
@@ -208,12 +209,9 @@ public ModulesConfigResponse getModules(@RequestParam("entity") String entityNam
...
 		Permission pluginPermission = null;
-		if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.WRITE))
-			pluginPermission = Permission.WRITE;
-		else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.READ))
-			pluginPermission = Permission.READ;
-		else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.COUNT))
-			pluginPermission = Permission.COUNT;
+		if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.WRITE)) pluginPermission = Permission.WRITE;
+		else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.READ)) pluginPermission = Permission.READ;
+		else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.COUNT)) pluginPermission = Permission.COUNT;
```
Comment (bartcharbon) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java]: Formatted the code, no functional change...

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java [hunk 2]
```diff
@@ -253,16 +251,16 @@ else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.
...
 					}
 					if (modReports)
 					{
 						String modEntitiesReportName = dataExplorerSettings.getEntityReport(entityName);
 						if (modEntitiesReportName != null)
 						{
-							modulesConfig
-									.add(new ModuleConfig("entitiesreport", modEntitiesReportName, "report-icon.png"));
+							modulesConfig.add(new ModuleConfig("entitiesreport", modEntitiesReportName,
+									"report-icon.png"));
```
Comment (bartcharbon) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java]: Formatted the code, no functional change...

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java [hunk 3]
```diff
@@ -301,10 +299,10 @@ else if (molgenisPermissionService.hasPermissionOnEntity(entityName, Permission.
...
 	private boolean isGenomeBrowserEntity(EntityMetaData entityMetaData)
 	{
-		AttributeMetaData attributeStartPosition = genomicDataSettings
-				.getAttributeMetadataForAttributeNameArray(GenomicDataSettings.Meta.ATTRS_POS, entityMetaData);
-		AttributeMetaData attributeChromosome = genomicDataSettings
-				.getAttributeMetadataForAttributeNameArray(GenomicDataSettings.Meta.ATTRS_CHROM, entityMetaData);
+		AttributeMetaData attributeStartPosition = genomicDataSettings.getAttributeMetadataForAttributeNameArray(
+				GenomicDataSettings.Meta.ATTRS_POS, entityMetaData);
+		AttributeMetaData attributeChromosome = genomicDataSettings.getAttributeMetadataForAttributeNameArray(
+				GenomicDataSettings.Meta.ATTRS_CHROM, entityMetaData);
```
Comment (bartcharbon) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java]: Formatted the code, no functional change...

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java [hunk 4]
```diff
@@ -532,6 +530,11 @@ private boolean viewExists(String viewName)
 		{
 			return freemarkerConfigurer.getConfiguration().getTemplate(viewName + ".ftl") != null;
 		}
+		catch (ParseException e)
+		{
+			LOG.info("error parsing template: ", e);
+			return false;
+		}
```
Comment (bartcharbon) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java]: This is the only "real" functional change

File: molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java (hunk 1 above)
Comment (fdlk) [inline molgenis-dataexplorer/src/main/java/org/molgenis/dataexplorer/controller/DataExplorerController.java]: :)

**Pull request merged**